lotus/client.go

```diff
@@ -61,7 +61,7 @@ func New(maddr ma.Multiaddr, authToken string) (*API, func(), error) {
 	closer, err := jsonrpc.NewMergeClient("ws://"+addr+"/rpc/v0", "Filecoin",
 		[]interface{}{
 			&api.Internal,
-		}, headers)
+		}, headers, jsonrpc.WithReconnect(true, time.Second*3, 0))
```

Comment:
Original PR that I did for lotus. You can configure a interval time to wait so that the websocket connection is recovered, adn the client can still use it as "nothing happened".

In the newer PR that will get merged, you can annotate with `retry:"true"` methods of the API. If that annotation is detected, the client retries itself the call until having success on a recovered connection. This makes the client even do less work since shouldn't do any kind of retry logic.

I may eventually consider switching to the new solution.